app/models/concerns/push_updates_for_subscription.rb

```diff
@@ -0,0 +1,46 @@
+#
+# The logic for who to send a message to is already encapsulated in the
+#   MessageSubscription module. This means the push update and in-app/mobile
+#   notification can be hooked in via the Subscription class.
+#   There is one subscription, per user, per message so send this push update
+#   direct to subscription.user_id 
+
+module PushUpdatesForSubscription
+  extend ActiveSupport::Concern
+
+  def update_client_store
+    data = { 
+      item: serialized_message,
+      sender: serialized_user,
+      operation: :create
+    }
+    channel = Channel.private_channel_for(self.user_id, app_name)
+    PushService.new.send_update_store(channel, app_name, data)
+  end
+
+  private
+
+  def serialized_message
+    associations = Message.reflections.keys.map(&:to_sym)
+    Api::V1::MessageSerializer.new(self.message, { exclude: associations })
+  end
+
+  # Determine which app to send it push update to.
+  # If recipient of message is also owner of its related_object, then send to donor/browse
+  # otherwise they are performing admin actions so send to Admin/Stock.
+  # This should work for admins logged in to DONOR app who create messages on their own offers.
+  def app_name
+    klass_name = self.related_object.class.name
```

Comment:
`related_object` is this a rails method ? or you forgot to define this method ?

Reply:
Good catch. Haven't written that method yet! I was thinking it could help us get round the lack of polymorphism in Message model. So...

```
def related_object
  offer || order
end
```

Reply:
Sure, that's a great approach, 
But as we have included this concern in Subscription class. `self` here would be the subscription's instance. So I think this method will be needed in `subscription.rb`
Please correct me if I'm wrong @steveyken .

Reply:
Well spotted. I switched to `self.message.related_object`